src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,245 @@
+        }
+        Scanner in = new Scanner(System.in);
+        while (runDuke) {
+            String userInput = in.nextLine();
+            try {
+                CommandType command = scanCommand(userInput);
+                switch (command) {
+                case BYE:
+                    runDuke = false;
+                    break;
+                case LIST:
+                    printList(tasks);
+                    break;
+                case DONE:
+                    int completedTaskNo = Integer.parseInt(userInput.replace("done ", ""));
+                    tasks.get(completedTaskNo - 1).markAsDone();
+                    printDoneResponse(tasks, completedTaskNo);
+                    updateData(tasks);
+                    break;
+                case DELETE:
+                    int deletedTaskNo = Integer.parseInt(userInput.replace("delete ", ""));
+                    printDeleteResponse(tasks, deletedTaskNo);
+                    tasks.remove(tasks.get(deletedTaskNo-1));
+                    updateData(tasks);
+                    break;
+                case TODO:
+                case EVENT:
+                case DEADLINE:
+                    addTask(tasks, userInput, command);
+                    printAddResponse(tasks, tasks.size());
+                    AddTasksasTxt(tasks, tasks.size() - 1);
+                    break;
+                }
+            } catch (InvalidCommandException e) {
+                printResponse(e.getMessage());
+            } catch (EmptyListException e) {
+                printResponse(e.getMessage());
+            } catch (NoDescriptionException e) {
+                printResponse(e.getMessage());
+            } catch (InvalidDescriptionException e) {
+                printResponse(e.getMessage());
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public static void AddTasksasTxt(ArrayList<Task> tasks, int index) throws IOException {
+        appendToFile(tasks.get(index).toString());
+    }
+
+    public static void updateData(ArrayList<Task> tasks) throws IOException {
+        FileWriter fw = new FileWriter("duke.txt");
+        fw.write("");
+        for (Task t : tasks) {
+            appendToFile(t.toString());
+        }
+    }
+
+    public static void appendToFile(String textToAppend) throws IOException {
+        FileWriter fw = new FileWriter("duke.txt", true);
+        fw.write(textToAppend + System.lineSeparator());
+        fw.close();
+    }
+
+    public static void loadPrevData(ArrayList<Task> tasks, String filePath) throws FileNotFoundException {
+        File f = new File(filePath);
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String data = s.nextLine();
+            char taskType = data.charAt(1);
+            String description = data.substring(7);
+            String detail;
+            String time;
+            switch (taskType) {
+            case 'T':
```

Comment:
Single characters may not be too useful in explaining what the task types actually are. See ['using name to explain'](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#use-name-to-explain) for suggestions.